### PR TITLE
added luarocks-tag-release job to release workflow action

### PR DESCRIPTION
**Pull request by gennaro-tedesco**

### CODE REVIEW

**Comment by gennaro-tedesco** [comment]
Good afternoon @mrcjkb, we spoke on reddit few days ago! This is an example of a plugin that uses a dependency, thus fulfilling the scope of creating luarocks dependencies spec.

Since this is the first time I am using your GitHub action, would you mind going over the change to make sure I have not forgotten anything needed?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,3 +14,17 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: LuaRocks Upload
+        uses: nvim-neorocks/luarocks-tag-release@v2.0.0
+        env:
+          LUAROCKS_API_KEY: ${{ secrets.LUAROCKS_API_KEY }}
+        with:
+          dependencies: |
+            fzf-lua
```
Comment (mrcjkb) [inline .github/workflows/release.yml:30]:
```suggestion
        with:
          dependencies: |
            fzf-lua
          copy_directories: |
            doc
```

Comment (mrcjkb) [inline .github/workflows/release.yml:30]: Also: `fzf-lua` is not on LuaRocks yet. I will open a PR to add it.

**Review by mrcjkb — COMMENTED** [review]
Nice :rocket: 

Have you added your `LUAROCKS_API_KEY` to this repo's GitHub actions secrets?

**Comment by mrcjkb** [comment]
We will have to wait for https://github.com/ibhagwan/fzf-lua/pull/640 before this will work.

**Comment by gennaro-tedesco** [comment]
> Have you added your `LUAROCKS_API_KEY` to this repo's GitHub actions secrets?

Yes, all is set, I would just need to create a new tag after merging and the action should trigger and push to the luarocks account :)

> We will have to wait for https://github.com/ibhagwan/fzf-lua/pull/640 before this will work.

Awesome, thank you very much for your help!

**Comment by mrcjkb** [comment]
`fzf-lua` is now on luarocks :tada: 
So we can proceed with this PR.

**Comment by gennaro-tedesco** [comment]
I am manually testing the action by adding a `worfklow_dispatch` event (so that the action can be triggered manually), however it [fails](https://github.com/gennaro-tedesco/nvim-possession/actions/runs/4209792208/jobs/7307009426) with
 ```
This GitHub workflow is designed to be run on tagged releases only
```
from [this comment](https://github.com/ibhagwan/fzf-lua/pull/640#issuecomment-1427151068) I take it should not be the case?

**Comment by gennaro-tedesco** [comment]
...after a few fixes...uh-uh, look at [this beauty](https://luarocks.org/modules/gennaro-tedesco/nvim-possession)!

**Comment by mrcjkb** [comment]
Nice! :partying_face: 

Some background information:
* The untagged release workflow only works if you set the version explicitly.
* The action tries to get the license info from GitHub, which is only possible if the license is on the main branch. So now that it's merged, it should be able to fetch it automatically.

One thing you might want to consider: `fzf-lua` is licensed under AGPL-3.0, which is a copyleft license (i.e. software that depends on it should be licensed under the same terms).
I don't think an MIT license is compatible with that.

**Pull request merged**